hazelcast/src/test/java/com/hazelcast/concurrent/semaphore/SemaphoreAdvancedTest.java

```diff
@@ -77,73 +82,63 @@ public void testSemaphoreWithFailures() throws InterruptedException {
         }
     }
 
-    @Test(timeout = 300000)
+    @Test(timeout = 5 * MINUTE)
     public void testSemaphoreWithFailuresAndJoin() {
         final String semaphoreName = randomString();
+        final Config config = new Config().setProperty(GroupProperty.PARTITION_COUNT.getName(), "5");
         final TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory(3);
-        final HazelcastInstance instance1 = factory.newHazelcastInstance();
-        final HazelcastInstance instance2 = factory.newHazelcastInstance();
-        final ISemaphore semaphore = instance1.getSemaphore(semaphoreName);
+        final HazelcastInstance instance1 = factory.newHazelcastInstance(config);
+        final HazelcastInstance instance2 = factory.newHazelcastInstance(config);
+        final ISemaphore semaphore1 = instance1.getSemaphore(semaphoreName);
         final CountDownLatch countDownLatch = new CountDownLatch(1);
 
-        assertTrue(semaphore.init(0));
+        assertTrue(semaphore1.init(0));
 
-        final Thread thread = new Thread() {
+        spawn(new Runnable() {
+            @Override
             public void run() {
                 for (int i = 0; i < 2; i++) {
                     try {
-                        semaphore.acquire();
+                        semaphore1.acquire();
                     } catch (InterruptedException e) {
                         e.printStackTrace();
                     }
                 }
                 countDownLatch.countDown();
             }
-        };
-        thread.start();
+        });
 
         instance2.shutdown();
-        semaphore.release();
-
-        HazelcastInstance instance3 = factory.newHazelcastInstance();
-        ISemaphore semaphore1 = instance3.getSemaphore(semaphoreName);
         semaphore1.release();
 
+        final HazelcastInstance instance3 = factory.newHazelcastInstance(config);
+        final ISemaphore semaphore3 = instance3.getSemaphore(semaphoreName);
+        semaphore3.release();
+
         assertOpenEventually(countDownLatch);
     }
 
-    @Test(timeout = 300000)
+    @Test(timeout = 5 * MINUTE)
     public void testMutex() throws InterruptedException {
         final String semaphoreName = randomString();
         final int threadCount = 2;
         final HazelcastInstance[] instances = createHazelcastInstanceFactory(threadCount).newInstances();
         final CountDownLatch latch = new CountDownLatch(threadCount);
         final int loopCount = 1000;
-
-        class Counter {
```

Comment:
Switched from this local class to an array with one element.

Reply:
You could also use `MutableInteger` or `MutableLong` here.